platforms/allwinner-d1/d1-core/src/timer.rs

```diff
@@ -2,12 +2,41 @@ pub use d1_pac::timer::tmr_ctrl::{
     TMR_CLK_PRES_A as TimerPrescaler, TMR_CLK_SRC_A as TimerSource, TMR_MODE_A as TimerMode,
 };
 use d1_pac::TIMER;
+use kernel::maitake::time::Clock;
 
 pub struct Timers {
     pub timer0: Timer0,
     pub timer1: Timer1,
 }
 
+impl Timer0 {
+    pub fn into_maitake_clock(mut self) -> Clock {
+        self.set_prescaler(TimerPrescaler::P8); // 24M / 8:  3.00M ticks/s
+        self.set_mode(TimerMode::PERIODIC);
```

Comment:
Same prescaler and mode configuration we set in the older code, so that's presumably not the problem...

Reply:
UPDATE: indeed it was not! https://github.com/tosc-rs/mnemos/pull/343#issuecomment-2563944528